lib/src/objdetect/objdetect_async.dart

```diff
@@ -0,0 +1,912 @@
+        (callback) =>
+            CFFI.CascadeClassifier_getFeatureType_Async(ref, callback), (c, p) {
+      final rval = p.cast<ffi.Int>().value;
+      return c.complete(rval);
+    });
+    return rval;
+  }
+
+  Future<(int, int)> getOriginalWindowSizeAsync() async {
+    final p = calloc<cvg.Size>();
+    final rval = cvRunAsync<(int, int)>(
+        (callback) =>
+            CFFI.CascadeClassifier_getOriginalWindowSize_Async(ref, callback),
+        (c, _) {
+      final ret = (p.ref.width, p.ref.height);
+      calloc.free(p);
+      return c.complete(ret);
+    });
+    return rval;
+  }
+
+  Future<bool> isOldFormatCascadeAsync() async {
+    final rval = cvRunAsync<bool>(
+        (callback) =>
+            CFFI.CascadeClassifier_isOldFormatCascade_Async(ref, callback),
+        (c, p) {
+      final rval = p.cast<ffi.Bool>().value;
+      return c.complete(rval);
+    });
+    return rval;
+  }
+
+  @override
+  cvg.CascadeClassifier get ref => ptr.ref;
+  static final finalizer = OcvFinalizer<cvg.CascadeClassifierPtr>(
+    CFFI.addresses.CascadeClassifier_Close,
+  );
+
+  void dispose() {
+    finalizer.detach(this);
+    CFFI.CascadeClassifier_Close(ptr);
+  }
+
+  @override
+  List<int> get props => [ptr.address];
+}
+
+class HOGDescriptor extends CvStruct<cvg.HOGDescriptor> {
+  HOGDescriptor._(cvg.HOGDescriptorPtr ptr, [bool attach = true])
+      : super.fromPointer(ptr) {
+    if (attach) {
+      finalizer.attach(this, ptr.cast(), detach: this);
+    }
+  }
+
+  factory HOGDescriptor.empty() {
+    final p = calloc<cvg.HOGDescriptor>();
+    cvRun(() => CFFI.HOGDescriptor_New(p));
+    return HOGDescriptor._(p);
+  }
+
+  factory HOGDescriptor.fromFile(String filename) {
+    final p = calloc<cvg.HOGDescriptor>();
+    final cp = filename.toNativeUtf8().cast<ffi.Char>();
+    cvRun(() => CFFI.HOGDescriptor_NewFromFile(cp, p));
+    calloc.free(cp);
+    return HOGDescriptor._(p);
+  }
+
+  Future<bool> loadAsync(String name) async {
+    final cname = name.toNativeUtf8().cast<ffi.Char>();
+    final rval = cvRunAsync<bool>(
+        (callback) => CFFI.HOGDescriptor_Load_Async(ref, cname, callback),
+        (c, p) {
+      final rval = p.cast<ffi.Bool>().value;
+      calloc.free(cname);
+      return c.complete(rval);
+    });
+    return rval;
+  }
+
+  Future<(VecFloat descriptors, VecPoint locations)> computeAsync(
+    Mat img, {
+    (int, int) winStride = (0, 0),
+    (int, int) padding = (0, 0),
+  }) async {
+    final descriptors = calloc<cvg.VecFloat>();
+    final locations = calloc<cvg.VecPoint>();
+    final rval = cvRunAsync2<(VecFloat, VecPoint)>(
+        (callback) => CFFI.HOGDescriptor_Compute_Async(
+              ref,
+              img.ref,
+              winStride.cvd.ref,
+              padding.cvd.ref,
+              callback,
+            ), (c, _, __) {
+      return c.complete(
+        (VecFloat.fromPointer(descriptors), VecPoint.fromPointer(locations)),
+      );
```

Comment:
same as before, descriptors and locations shouldn't be created here, use the pointers provided by onComplete,
```dart
(c, p, p1) =>
      c.complete(
        (VecFloat.fromPointer(p.cast<cvg.VecFloat>()), VecPoint.fromPointer(p1.cast<cvg.VecPoint>())),
      )
```